src/print.jl

```diff
@@ -23,8 +23,25 @@
 #############################################################################
 
 # Used for dispatching
+"""
+    PrintMode
+
+An abstract type used for dispatching printing.
+"""
 abstract type PrintMode end
+
+"""
+    REPLMode
+
+A type used for dispatching printing. Produces plain-text representations.
+"""
 abstract type REPLMode <: PrintMode end
```

Comment:
A few questions on these. Why are the abstract? Is anyone subtyping them? Ideally, for JuMP 1.0 we would replace and use `MIME"text/plain"` and `MIME"text/latex"` instead.